test/integration_test.py

```diff
@@ -889,135 +874,196 @@ def _list_bundles(self,
-                    # However, we still shouldn't observe any files that don't
-                    # occur in *any* of the expected bundles.
-                    found_bundles = row_bundles & expected_bundles
-                    self.assertNotEqual(set(), found_bundles)
-                    all_found_bundles.update(found_bundles)
-                self.assertEqual(expected_bundles, all_found_bundles)
-
-            # With authorized credentials, all bundles included in the filters
-            # should be represented in the manifest
-            with self._service_account_credentials:
-                assert_manifest({public_bundle, *managed_access_bundles})
-
-            # Without credentials, only the public bundle should be represented
-            assert_manifest({public_bundle})
-
-            # Create a single-file curl manifest and verify that the OAuth2
-            # token is present in the .curlrc
-            managed_access_file = self.random.choice(self.random.choice(hits)['files'])
-            manifest_url.set(args={
-                'filters': json.dumps({'fileId': {'is': [managed_access_file['uuid']]}}),
-                'format': 'curl'
-            })
-            with self._service_account_credentials:
-                response = self._get_url_content(str(manifest_url))
-            self._check_curl_manifest(catalog, response)
-            auth_header_line = one(
-                line
-                for line in TextIOWrapper(BytesIO(response))
-                if line.startswith('--header')
-            )
-            auth_header = auth_header_line.split(' ', 1).pop().strip('"\n')
-            self.assertEqual(f'Authorization: Bearer {self._tdr_client.credentials.token}',
-                             auth_header)
+        source_filter = {'sourceId': {'is': list(managed_access_source_ids)}}
+        url = furl(config.service_endpoint(),
+                   path='/index/bundles',
+                   args={'filters': json.dumps(source_filter)})
+        response = self._get_url_unchecked(str(url))
+        self.assertEqual(response.status, 403 if managed_access_source_ids else 200)
+
+        with self._service_account_credentials:
+            hits = self._get_entities(catalog, 'bundles', filters=source_filter)
+        hit_source_ids = source_ids_from_hits(hits)
+        self.assertEqual(hit_source_ids, managed_access_source_ids)
+        return hits
+
+    def _test_managed_access_repository_files(self, bundles: JSONs) -> Set[str]:
+        """
+        Test the managed access controls for the /repository/files endpoint
+        :return: download URLs for the managed access files
+        """
+        managed_access_file_urls = {
+            file['url']
+            for bundle in bundles
+            for file in bundle['files']
+        }
+        file_url = first(managed_access_file_urls)
+        response = self._get_url_unchecked(file_url, redirect=False)
+        self.assertEqual(response.status, 404)
```

Comment:
I believe the convention with assertions is expected followed by actual.

Reply:
I found an instance of this convention being suggested by Hannes https://github.com/DataBiosphere/azul/pull/3461#discussion_r727636605.

Reply:
> I would've liked to see the penultimate commit as a split commit, one with the refactoring and the second with the breakup of the tests

I'm not 100% sure what "refactoring" you're to, since the entirety of the main commit is a refactoring, but I did what I could.